velox/functions/sparksql/aggregates/CovarianceAggregate.cpp

```diff
@@ -47,24 +47,31 @@ struct CovarSampResultAccessor {
 template <bool nullOnDivideByZero>
 struct CorrResultAccessor {
   static bool hasResult(const CorrAccumulator& accumulator) {
+    bool hasNaN = std::isnan(accumulator.m2Y()) ||
+        std::isnan(accumulator.m2X()) || std::isnan(accumulator.c2());
+    bool notZero = accumulator.m2Y() != 0 && accumulator.m2X() != 0;
     if constexpr (nullOnDivideByZero) {
       if (accumulator.count() == 1) {
         return false;
       }
-      return accumulator.m2X() != 0 && accumulator.m2Y() != 0;
     } else {
       if (accumulator.count() == 1) {
         return true;
       }
-      return accumulator.m2X() != 0 && accumulator.m2Y() != 0;
     }
+    return hasNaN || notZero;
   }
 
   static double result(const CorrAccumulator& accumulator) {
     if (accumulator.count() == 1) {
```

Comment:
nit: 
```
if constexpr (!nullOnDivideByZero) {
if (accumulator.count() == 1) {
}
}
```

Reply:
For the case where `nullOnDivideByZero` is true and `accumulator.count() == 1`, the `hasResult` returns false and `result` should not be called. This check is to ensure that so `if constexpr (!nullOnDivideByZero)` is not added. Do you think it makes sense?

Reply:
Only adjust the order, move nullOnDivideByZero first.

Reply:
This looks a bit complicated, I would suggest keep current implementation
if constexpr (!nullOnDivideByZero) {
if (count == 1) {
return NAN
} else {
exception
}
} else {
if (count == 1) {
}